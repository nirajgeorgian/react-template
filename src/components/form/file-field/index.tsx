import Dragger, { DraggerProps } from 'antd/lib/upload/Dragger'

import { FieldProps } from 'formik/dist/Field'
import Icon from 'antd/lib/icon'
import React from 'react'
import { UploadChangeParam } from 'antd/lib/upload'

interface _ICreateAntFilesFieldProps extends DraggerProps, FieldProps {}
export const CreateAntFilesField: React.FC<_ICreateAntFilesFieldProps> = ({ form, field }) => {
	const fileProps = {
		name: field.name,
		beforeUpload: () => false,
		onChange: (files: UploadChangeParam) => {
			form.setFieldValue(field.name, [...files.fileList])
		},
		multiple: true
	}

	return (
		<Dragger {...fileProps}>
			<p className="ant-upload-drag-icon">
				<Icon type="inbox" />
			</p>
			<p className="ant-upload-text">Click or drag file to this area to upload</p>
			<p className="ant-upload-hint">
				Support for a single or bulk upload. Strictly prohibit from uploading company data or other band files
			</p>
		</Dragger>
	)
}
