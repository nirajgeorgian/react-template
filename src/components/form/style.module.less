.title_select_option {
	div {
		height: 3rem;
	}
}

// --------- stayology-select ----------------
.stayology_select_outer {
	display: flex;
	flex-direction: column;
	box-shadow: none !important;
	

	div {
		box-shadow: none !important;
		width: 100%;
	}
}

.satyology_select_header {
	
	text-align: left;
	// margin-left: 9%;
	line-height: 100%;
}

.satyology_select_footer {
	text-align: left;
	// margin-left: 9%;
}


.satyology_select {
	display: flex;
}

.satyology_select_source {
	display: flex;
	flex-direction: column;
	margin: 2%;
	font-size: 2rem;
}

.satyology_select_icon {
	margin: 2%;

	img {
		width: 7rem;
	}
}

.satyology_select_destination {
	display: flex;
	flex-direction: column;
	margin: 2%;
	font-size: 2rem;
	border: 0 !important;
}

// --------- stayology-button ----------------
.stayology_button {
	box-shadow: none !important;
	color: #fff !important;

	&:hover,
	&:focus {
		border-color: #d9d9d9 !important;
	}

	span {
		text-shadow: none !important;
	}
}

.stayology_button_primary {
	background-color: #32eb28 !important;
	border-color: #32eb28 !important;
	border-radius: 0;

	&:hover,
	&:focus {
		border-color: #32eb28 !important;
	}
}

.stayology_button_secondary {
	background-color: #0a0a0a !important;
	color: blanchedalmond !important;
	border-color: #ffffff !important;
	border-radius: 0;

	&:hover,
	&:focus {
		border-color: #0a0a0a !important;
	}
}

// ---------------------Event Tile--------------------

.EventTile {
	// color: black;
	// background-color: #F5F5F5;
	// padding: 2.5%;
	// height: 100%;
	// width: 100%;
	// display: flex;
	// flex-direction: row;
	// justify-content: space-between;
	// border-radius: 4px;
	font-family: 'Lato', sans-serif;
}

.EventTile_card {
	box-shadow: 0px 0px 3px rgba(132, 136, 136, 0.342) !important;
	border-radius: 4;
	width: 100%;
	height: 290px !important;
	background-color: #ffffff 0% 0% no-repeat padding-box;
	&:hover,
	&:focus {
		background-color: #f5f5f5;
	}

	div {
		padding-top: 3px !important;
	}
}

.EventTile_card_outerdiv {
	width: 100%;
	display: flex;
	flex-direction: row;
	// padding: 2px !important;
}

.EventTile_league_logo_div {
	width: 2%;
}

.EventTile_league_logo {
	width: 2.5rem;
}

.EventTile_card_content {
	width: 100%;
	padding: '3%';
	// margin
}

.EventTile_card_team_name {
	font-size: 1.1rem;
	font-weight: 300;
	text-align: center;
}

.EventTile_card_location {
	font-size: 0.7rem;
	font-weight: 300;
	line-height: 0.3rem;
}

.EventTile_price {
	text-align: center;
	margin-top: 3%;
	font-weight: 600;
	color: black
}

.EventTile_teams {
	width: 20%;
	display: flex;
	flex-direction: row;
	font-family: 'Lato', sans-serif;
	font-size: 1rem;
	font-weight: 400;

	div {
		img {
			width: 1.8rem;
		}
	}
}

.EventTile_datetime {
	margin-bottom: 10%;
	width: 15%;
}

.EventTile_tag {
	border-radius: 0px !important;
	box-shadow: none !important;

	span {
		color: black;
		font-weight: 300;
		font-weight: bold;
	}
}

.EventTile_time {
	width: 100%;
	text-align: center;
}

.EventTile_date {
	width: 88%;
	text-align: center;
	font-size: 0.8rem;
}

.EventTile_location {
	text-align: left;
	font-family: 'Lato', sans-serif;
	width: 25%;
}

.EventTile_country {
	width: 100%;
	text-align: left;
	font-size: 0.8rem;
	color: grey;
}

.EventTile_button {
	width: 15%;

	button {
		border-radius: 4px !important;
		border: none !important;
		box-shadow: 0px 4px 10px rgba(132, 136, 136, 0.342) !important;
	}
}

.EventTile_width33 {
	width: 33.33%;
}

//////f1 event tile

.f1_EventTile_league_logo_div {
	width: 2%;
}
.f1_EventTile_card_team_name {
	margin-top: 10%;
	font-size: 1rem;
	font-weight: 800;
	text-align: center;
	color:black
}
.f1_EventTile_card_location {
	font-size: 0.7rem;
	font-weight: 300;
	line-height: 0.3rem;
}

// -------------------.EventTile Tile responsive-------------------------

// @media only screen and (max-width:1151px){
// 	.EventTile{
// 		color: black;
// 		background-color: #F5F5F5;
// 		padding: 4%;
// 		height: 100%;
// 		width: 100%;
// 		display: flex;
// 		flex-direction: row;
// 	}
// 	.EventTile_teams{
// 		width: 20%;
// 		display: flex;
// 		flex-direction: row;
// 		font-family: 'Lato', sans-serif;
// 		font-size: 0.9rem;
// 		font-weight: 400;

// 		div{
// 			img{
// 				width: 1.3rem;
// 			}
// 		}
// 	}

// .EventTile_button{
// 	width: 15%;
// 	button{
// 		// width:60px;
// 		border-radius: 4px !important;
// 		border: none !important;
// 		box-shadow: 0px 4px 10px rgba(132, 136, 136, 0.342)  !important;

// 	}

// }

// }
// @media only screen and (max-width:1151px){
// 	.EventTile{
// 		color: black;
// 		background-color: #F5F5F5;
// 		padding: 4.5%;
// 		height: 100%;
// 		width: 100%;
// 		display: flex;
// 		flex-direction: row;
// 	}
// 	.EventTile_teams{
// 		width: 30%;
// 		display: flex;
// 		flex-direction: row;
// 		font-family: 'Lato', sans-serif;
// 		font-size: 1rem;
// 		font-weight: 400;

// 		div{
// 			img{
// 				width: 1.2rem;
// 			}
// 		}
// 	}

// .EventTile_button{
// 	width: 20%;
// 	button{
// 		// width:60px;
// 		border-radius: 4px !important;
// 		border: none !important;
// 		box-shadow: 0px 4px 10px rgba(132, 136, 136, 0.342)  !important;

// 	}

// }

// }
// @media only screen and (max-width:995px){
// 	.EventTile{
// 		color: black;
// 		background-color: #F5F5F5;
// 		padding: 4.5%;
// 		height: 100%;
// 		width: 100%;
// 		display: flex;
// 		flex-direction: row;
// 	}
// 	.EventTile_teams{
// 		width: 30%;
// 		display: flex;
// 		flex-direction: row;
// 		font-family: 'Lato', sans-serif;
// 		font-size: 0.8rem;
// 		font-weight: 400;

// 		div{
// 			img{
// 				width: 1.2rem;
// 			}
// 		}
// 	}
// 	.EventTile_tag{
// 		span{
// 		font-size: 0.6rem;;
// 		}
// 	}
// 	.EventTile_date{
// 		width: 88%;
// 		text-align: center;
// 		font-size: 0.6rem;
// 	}
// 	.EventTile_location{
// 		text-align: left;
// 		font-family: 'Lato', sans-serif;
// 		width: 25%;
// 		font-size: 0.7rem;
// 	}

// 	.EventTile_country{
// 		width: 100%;
// 		text-align: left;
// 		font-size: 0.5rem;
// 		color: grey;
// 	}

// .EventTile_button{
// 	width: 15%;
// 	button{
// 		// width:60px;
// 		border-radius: 4px !important;
// 		font-size: 0.6rem;
// 		border: none !important;
// 		box-shadow: 0px 4px 10px rgba(132, 136, 136, 0.342)  !important;

// 	}

// }

// }
// @media only screen and (max-width:855px){
// 	.EventTile{
// 		color: black;
// 		background-color: #F5F5F5;
// 		padding: 5%;
// 		height: 100%;
// 		width: 100%;
// 		display: flex;
// 		flex-direction: row;
// 	}
// 	.EventTile_teams{
// 		width: 30%;
// 		display: flex;
// 		flex-direction: row;
// 		font-family: 'Lato', sans-serif;
// 		font-size: 0.6rem;
// 		font-weight: 400;

// 		div{
// 			img{
// 				width: 1rem;
// 			}
// 		}
// 	}
// 	.EventTile_tag{
// 		span{
// 		font-size: 0.5rem;;
// 		}
// 	}
// 	.EventTile_date{
// 		width: 88%;
// 		text-align: center;
// 		font-size: 0.5rem;
// 	}
// 	.EventTile_location{
// 		text-align: left;
// 		font-family: 'Lato', sans-serif;
// 		width: 25%;
// 		font-size: 0.6rem;
// 	}

// 	.EventTile_country{
// 		width: 100%;
// 		text-align: left;
// 		font-size: 0.4rem;
// 		color: grey;
// 	}

// .EventTile_button{
// 	width: 15%;
// 	button{
// 		// width:60px;
// 		border-radius: 4px !important;
// 		font-size: 0.5rem;
// 		border: none !important;
// 		box-shadow: 0px 4px 10px rgba(132, 136, 136, 0.342)  !important;

// 	}

// }

// }
