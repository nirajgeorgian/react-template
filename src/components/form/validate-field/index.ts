export const validateEmail = (value: string) => {
	let errors

	if (!value) {
		errors = 'Required!'
	} else if (!/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(value)) {
		errors = 'Invalid email address!'
	}

	return errors
}

export const validatePhone = (value: string) => {
	let errors
	if (!value) {
		errors = 'Required!'
	} else if (!/^[0-9]{10}$/.test(value)) {
		errors = 'Invalid phone number!'
	}

	return errors
}

export const isRequired = (value: string) => (!value ? 'Required!' : '')
