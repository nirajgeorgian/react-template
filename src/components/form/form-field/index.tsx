import { AutoComplete, Checkbox, DatePicker, Form, Input, Radio, Select, TimePicker } from 'antd'
import { FieldProps, FormikProps } from 'formik'

import { FormItemProps } from 'antd/lib/form/FormItem'
import React from 'react'

const FormItem = Form.Item
const { Option } = Select

interface _ICreateAntAntFieldProps extends FieldProps, FormItemProps {
	type: string
	icon?: string
	radioOptions?: Array<string>
	selectOptions?: Array<string>
	simple?: boolean
	submitCount?: number
}
const CreateAntField = (AntComponent: React.ComponentType<any>): React.FC<_ICreateAntAntFieldProps> => ({
	field,
	form,
	hasFeedback,
	label,
	icon,
	selectOptions,
	radioOptions,
	simple = true,
	submitCount,
	type = 'text',
	...props
}): React.ReactElement => {
	const touched = form.touched[field.name]
	const submitted = submitCount && submitCount > 0

	const hasError = form.errors[field.name]
	const submittedError = hasError && submitted
	const touchedError = hasError && touched

	const onInputChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) =>
		form.setFieldValue(field.name, value)
	const onCheckboxChange = ({ target: { checked } }: React.ChangeEvent<HTMLInputElement>) =>
		form.setFieldValue(field.name, checked)
	const onChange = (value: FormikProps<any>) => form.setFieldValue(field.name, value)
	const onBlur = () => form.setFieldTouched(field.name, true)

	const renderOption = (option: string) => <Option key={option}>{option}</Option>

	const renderRadio = (radio: string) => (
		<Radio key={radio} value={radio}>
			{radio}
		</Radio>
	)
	const selectOptionsRender = selectOptions ? selectOptions.map(renderOption) : null
	const radioOptionsRender = radioOptions ? radioOptions.map(renderRadio) : null

	return (
		<div className="field-container">
			<FormItem
				className={props.className ? props.className : ''}
				label={label}
				help={submittedError || touchedError ? hasError : false}
				validateStatus={submittedError || touchedError ? 'error' : 'success'}>
				{selectOptionsRender && (
					<AntComponent
						{...field}
						{...props}
						type={type}
						autoComplete="on"
						onBlur={onBlur}
						onChange={type ? onInputChange : onChange}>
						{selectOptionsRender}
					</AntComponent>
				)}
				{radioOptionsRender && (
					<AntComponent
						{...field}
						{...props}
						type={type}
						autoComplete="off"
						radioOptions={['option1', 'option2']}
						onBlur={onBlur}
						onChange={type ? onInputChange : onChange}>
						{radioOptionsRender}
					</AntComponent>
				)}
				{type === 'checkbox' && (
					<AntComponent
						{...field}
						{...props}
						type={type}
						autoComplete="off"
						onBlur={onBlur}
						onChange={onCheckboxChange}
					/>
				)}
				{!selectOptionsRender && !radioOptionsRender && simple && (
					<AntComponent
						{...field}
						{...props}
						type={type}
						autoComplete="off"
						onBlur={onBlur}
						onChange={type ? onInputChange : onChange}
					/>
				)}
			</FormItem>
		</div>
	)
}

export const AntSelect = CreateAntField(Select)
export const AntDatePicker = CreateAntField(DatePicker)
export const AntDateRangePicker = CreateAntField(DatePicker.RangePicker)
export const AntInput = CreateAntField(Input)
export const AntPassword = CreateAntField(Input.Password)
export const AntTimePicker = CreateAntField(TimePicker)
export const AntRadio = CreateAntField(Radio.Group)
export const AntAutoComplete = CreateAntField(AutoComplete)
export const AntCheckbox = CreateAntField(Checkbox)
