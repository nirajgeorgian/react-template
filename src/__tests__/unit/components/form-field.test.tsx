import '@testing-library/jest-dom/extend-expect';

import { Field, Form, Formik, FormikValues } from 'formik';
import { fireEvent, render, wait } from '@testing-library/react';

import { AntInput } from 'components/form/form-field'
import Button from 'antd/lib/button';
import React from 'react';

const AntInputComponent = (props: FormikValues): JSX.Element => {
	const { values } = props

	return (
		<Form>
			<Field component={AntInput} id="username" label="username" name="username" value={values.username || ''} />
			<Field component={AntInput} id="password" label="password" name="password" value={values.password || ''} />
			<Button htmlType="submit">
				submit
			</Button>
		</Form>
	)
}

const setup = () => {
	const handleSubmit = jest.fn()
	const handleValidate = jest.fn(() => { })

	const { getByLabelText, getByText, container, debug, ...utils } = render(
		<Formik initialValues={{}} validate={handleValidate} render={AntInputComponent} onSubmit={handleSubmit} />
	)

	const user = { username: 'dodo', password: 'dododuck' }
	const username = container.querySelector('input[name="username"]');
	const password = container.querySelector('input[name="password"]');
	const submit = container.querySelector('button[type="submit"]');

	const changeUsernameInput = (value: string) => username && fireEvent.change(username, { target: { value } })
	const changePasswordInput = (value: string) => password && fireEvent.change(password, { target: { value } })
	const clickSubmit = () => submit && fireEvent.click(submit)

	return {
		...utils,
		handleSubmit,
		user,
		changeUsernameInput,
		changePasswordInput,
		clickSubmit,
	}
}

const setupSuccessCase = async () => {
	const utils = setup()
	utils.changeUsernameInput(utils.user.username)
	utils.changePasswordInput(utils.user.password)
	await wait(() => utils.clickSubmit())

	return utils
}
const setupWithNoPassword = () => {
	const utils = setup()
	utils.changeUsernameInput(utils.user.username)
	utils.clickSubmit()
	const errorMessage = utils.getByRole('alert')

	return { ...utils, errorMessage }
}
const setupWithNoUsername = () => {
	const utils = setup()
	utils.changePasswordInput(utils.user.password)
	utils.clickSubmit()
	const errorMessage = utils.getByRole('alert')

	return { ...utils, errorMessage }
}

// test cases in sequence
test("AntInput field render's properly with <Formik>", () => {
	const onSubmit = jest.fn();
	const { asFragment } = render(
		<Formik initialValues={{}} render={AntInputComponent} onSubmit={onSubmit} />
	)
	expect(asFragment()).toMatchSnapshot()
})
test('calls onSubmit with the username and password', async () => {
	const { handleSubmit } = await setupSuccessCase()

	await wait(() => expect(handleSubmit).toHaveBeenCalledTimes(1))
})