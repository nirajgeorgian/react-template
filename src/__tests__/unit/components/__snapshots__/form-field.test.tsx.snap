// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AntInput field render's properly with <Formik> 1`] = `
<DocumentFragment>
  <form>
    <div
      class="field-container"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            title="username"
          >
            username
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control-wrapper"
        >
          <div
            class="ant-form-item-control has-success"
          >
            <span
              class="ant-form-item-children"
            >
              <input
                autocomplete="off"
                class="ant-input"
                id="username"
                name="username"
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="field-container"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            title="password"
          >
            password
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control-wrapper"
        >
          <div
            class="ant-form-item-control has-success"
          >
            <span
              class="ant-form-item-children"
            >
              <input
                autocomplete="off"
                class="ant-input"
                id="password"
                name="password"
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <button
      class="ant-btn"
      type="submit"
    >
      <span>
        submit
      </span>
    </button>
  </form>
</DocumentFragment>
`;
